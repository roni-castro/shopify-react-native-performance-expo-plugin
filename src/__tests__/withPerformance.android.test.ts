import {
  addShopifyPerformanceData,
  addShopifyPerformanceImport,
  addShopifyPerformanceInitialization,
} from "../withPerformance.android";
import { getFixture } from "./fixtures/getFixture";

describe("addShopifyPerformanceInitialization", () => {
  describe("MainApplication.kt", () => {
    const kotlinMainApplication = getFixture("MainApplication.kt");

    it("adds the Shopify Performance initialization and import to onCreate method", () => {
      const result = addShopifyPerformanceData(kotlinMainApplication, false);

      expect(result).toMatchSnapshot();
    });

    it("adds the Shopify Performance initialization to onCreate method", () => {
      const results = addShopifyPerformanceInitialization(
        kotlinMainApplication,
        false,
      );

      expect(results.contents).toMatch(
        "ReactNativePerformance.onAppStarted()\n",
      );
      // did add new content
      expect(results.didMerge).toBe(true);
      // didn't remove old content
      expect(results.didClear).toBe(false);
    });

    it("adds the Shopify Performance import", () => {
      const result = addShopifyPerformanceImport(kotlinMainApplication, false);

      expect(result).toMatch(
        "com.shopify.reactnativeperformance.ReactNativePerformance",
      );
    });
  });

  describe("MainApplication.java", () => {
    const javaMainApplication = getFixture("MainApplication.java");

    it("adds the Shopify Performance initialization and import to onCreate method", () => {
      const result = addShopifyPerformanceData(javaMainApplication, true);

      expect(result).toMatchSnapshot();
    });

    it("adds the initialization to onCreate method", () => {
      const results = addShopifyPerformanceInitialization(
        javaMainApplication,
        true,
      );

      expect(results.contents).toMatch(
        "ReactNativePerformance.onAppStarted();\n",
      );
      // did add new content
      expect(results.didMerge).toBe(true);
      // didn't remove old content
      expect(results.didClear).toBe(false);
    });

    it("adds the Shopify Performance import", () => {
      const result = addShopifyPerformanceImport(javaMainApplication, true);

      expect(result).toMatch(
        "com.shopify.reactnativeperformance.ReactNativePerformance",
      );
    });
  });
});
