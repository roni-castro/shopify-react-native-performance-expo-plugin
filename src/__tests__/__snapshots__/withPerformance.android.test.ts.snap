// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addShopifyPerformanceInitialization MainApplication.java adds the Shopify Performance initialization and import to onCreate method 1`] = `
"package com.rndiffapp;
import com.shopify.reactnativeperformance.ReactNativePerformance;
 
import android.app.Application;
import android.content.Context;
import com.facebook.react.PackageList;
import com.facebook.react.ReactApplication;
import com.facebook.react.ReactInstanceManager;
import com.facebook.react.ReactNativeHost;
import com.facebook.react.ReactPackage;
import com.facebook.react.config.ReactFeatureFlags;
import com.facebook.soloader.SoLoader;
import com.rndiffapp.newarchitecture.MainApplicationReactNativeHost;
import java.lang.reflect.InvocationTargetException;
import java.util.List;
 
public class MainApplication extends Application implements ReactApplication {
 
  private final ReactNativeHost mReactNativeHost =
      new ReactNativeHost(this) {
        @Override
        public boolean getUseDeveloperSupport() {
          return BuildConfig.DEBUG;
        }
 
        @Override
        protected List<ReactPackage> getPackages() {
          @SuppressWarnings("UnnecessaryLocalVariable")
          List<ReactPackage> packages = new PackageList(this).getPackages();
          // Packages that cannot be autolinked yet can be added manually here, for example:
          // packages.add(new MyReactNativePackage());
          return packages;
        }
 
        @Override
        protected String getJSMainModuleName() {
          return "index";
        }
      };
 
  private final ReactNativeHost mNewArchitectureNativeHost =
      new MainApplicationReactNativeHost(this);
 
  @Override
  public ReactNativeHost getReactNativeHost() {
    if (BuildConfig.IS_NEW_ARCHITECTURE_ENABLED) {
      return mNewArchitectureNativeHost;
    } else {
      return mReactNativeHost;
    }
  }
 
  @Override
  public void onCreate() {
// @generated begin react-native-performance-onCreate-android - expo prebuild (DO NOT MODIFY) sync-1376c13cc9e8da676e76ccfb6d9d763e1b14d640
  	ReactNativePerformance.onAppStarted();

// @generated end react-native-performance-onCreate-android
    super.onCreate();
    // If you opted-in for the New Architecture, we enable the TurboModule system
    ReactFeatureFlags.useTurboModules = BuildConfig.IS_NEW_ARCHITECTURE_ENABLED;
    SoLoader.init(this, /* native exopackage */ false);
    initializeFlipper(this, getReactNativeHost().getReactInstanceManager());
  }
 
  /**
   * Loads Flipper in React Native templates. Call this in the onCreate method with something like
   * initializeFlipper(this, getReactNativeHost().getReactInstanceManager());
   *
   * @param context
   * @param reactInstanceManager
   */
  private static void initializeFlipper(
      Context context, ReactInstanceManager reactInstanceManager) {
    if (BuildConfig.DEBUG) {
      try {
        /*
         We use reflection here to pick up the class that initializes Flipper,
        since Flipper library is not available in release mode
        */
        Class<?> aClass = Class.forName("com.rndiffapp.ReactNativeFlipper");
        aClass
            .getMethod("initializeFlipper", Context.class, ReactInstanceManager.class)
            .invoke(null, context, reactInstanceManager);
      } catch (ClassNotFoundException e) {
        e.printStackTrace();
      } catch (NoSuchMethodException e) {
        e.printStackTrace();
      } catch (IllegalAccessException e) {
        e.printStackTrace();
      } catch (InvocationTargetException e) {
        e.printStackTrace();
      }
    }
  }
}"
`;

exports[`addShopifyPerformanceInitialization MainApplication.kt adds the Shopify Performance initialization and import to onCreate method 1`] = `
"package com.helloworld
import com.shopify.reactnativeperformance.ReactNativePerformance

import android.app.Application
import android.content.res.Configuration

import com.facebook.react.PackageList
import com.facebook.react.ReactApplication
import com.facebook.react.ReactNativeHost
import com.facebook.react.ReactPackage
import com.facebook.react.ReactHost
import com.facebook.react.defaults.DefaultNewArchitectureEntryPoint.load
import com.facebook.react.defaults.DefaultReactNativeHost
import com.facebook.react.soloader.OpenSourceMergedSoMapping
import com.facebook.soloader.SoLoader

import expo.modules.ApplicationLifecycleDispatcher
import expo.modules.ReactNativeHostWrapper

class MainApplication : Application(), ReactApplication {

  override val reactNativeHost: ReactNativeHost = ReactNativeHostWrapper(
        this,
        object : DefaultReactNativeHost(this) {
          override fun getPackages(): List<ReactPackage> {
            val packages = PackageList(this).packages
            // Packages that cannot be autolinked yet can be added manually here, for example:
            // packages.add(new MyReactNativePackage());
            return packages
          }

          override fun getJSMainModuleName(): String = ".expo/.virtual-metro-entry"

          override fun getUseDeveloperSupport(): Boolean = BuildConfig.DEBUG

          override val isNewArchEnabled: Boolean = BuildConfig.IS_NEW_ARCHITECTURE_ENABLED
          override val isHermesEnabled: Boolean = BuildConfig.IS_HERMES_ENABLED
      }
  )

  override val reactHost: ReactHost
    get() = ReactNativeHostWrapper.createReactHost(applicationContext, reactNativeHost)

  override fun onCreate() {
// @generated begin react-native-performance-onCreate-android - expo prebuild (DO NOT MODIFY) sync-4d31cd043c736494ae93b063c742108267af9cb6
  	ReactNativePerformance.onAppStarted()

// @generated end react-native-performance-onCreate-android
    super.onCreate()
    SoLoader.init(this, OpenSourceMergedSoMapping)
    if (BuildConfig.IS_NEW_ARCHITECTURE_ENABLED) {
      // If you opted-in for the New Architecture, we load the native entry point for this app.
      load()
    }
    ApplicationLifecycleDispatcher.onApplicationCreate(this)
  }

  override fun onConfigurationChanged(newConfig: Configuration) {
    super.onConfigurationChanged(newConfig)
    ApplicationLifecycleDispatcher.onConfigurationChanged(this, newConfig)
  }
}"
`;
